From 986e8df0d8e7555d09a23d32f3783099711d4726 Mon Sep 17 00:00:00 2001
Message-Id: <986e8df0d8e7555d09a23d32f3783099711d4726.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Sagar Kamble <sagar.a.kamble@intel.com>
Date: Mon, 25 Nov 2013 14:31:28 +0530
Subject: [PATCH 117/312] FOR_UPSTREAM [VPG]: drm/i915: Updating waits added
 in Punit read/write path to atomic waits

Currently wait for IOSF SB busy status in PUNIT read and write may cause
worker thread sleep which will lead to delayed execution of worker.
Changing the wait to atomic with this patch

Issue: VIZ-3565
Change-Id: Ifd3040ed6a63cb79198dedd3bbf3d63421be3bbf
Signed-off-by: Sagar Kamble <sagar.a.kamble@intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
Reviewed-on: https://android.intel.com/165494
Reviewed-by: Bloomfield, Jon <jon.bloomfield@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_sideband.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_sideband.c b/drivers/gpu/drm/i915/intel_sideband.c
index 7a3da0e..fd0cf1d 100644
--- a/drivers/gpu/drm/i915/intel_sideband.c
+++ b/drivers/gpu/drm/i915/intel_sideband.c
@@ -58,7 +58,8 @@ static int vlv_sideband_rw(struct drm_i915_private *dev_priv, u32 devfn,
 
 	WARN_ON(!mutex_is_locked(&dev_priv->dpio_lock));
 
-	if (wait_for((I915_READ(VLV_IOSF_DOORBELL_REQ) & IOSF_SB_BUSY) == 0, 5)) {
+	if (wait_for_atomic((I915_READ(VLV_IOSF_DOORBELL_REQ)
+					& IOSF_SB_BUSY) == 0, 5)) {
 		DRM_DEBUG_DRIVER("IOSF sideband idle wait (%s) timed out\n",
 				 is_read ? "read" : "write");
 		return -EAGAIN;
@@ -69,7 +70,8 @@ static int vlv_sideband_rw(struct drm_i915_private *dev_priv, u32 devfn,
 		I915_WRITE(VLV_IOSF_DATA, *val);
 	I915_WRITE(VLV_IOSF_DOORBELL_REQ, cmd);
 
-	if (wait_for((I915_READ(VLV_IOSF_DOORBELL_REQ) & IOSF_SB_BUSY) == 0, 5)) {
+	if (wait_for_atomic((I915_READ(VLV_IOSF_DOORBELL_REQ)
+					& IOSF_SB_BUSY) == 0, 5)) {
 		DRM_DEBUG_DRIVER("IOSF sideband finish wait (%s) timed out\n",
 				 is_read ? "read" : "write");
 		return -ETIMEDOUT;
-- 
1.7.9.5

